### PR TITLE
add l2 metric aliases

### LINKED ISSUE #1602: LightGBMTunerCV not working for regression objective
The script https://github.com/optuna/optuna/blob/master/examples/lightgbm_tuner_cv.py runs just fine as it is. However, I get a `KeyError: 'mse-mean'` if I change the `objective` to `regression` and `metric` to `mse`. Similar erro happens to other metrics as well when the `objective` is set to `regression`.

## Environment

- Optuna version: 2.0.0
- Python version: 3.7
- OS: MacOS Catalina

## Error messages, stack traces, or logs

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-11-7753103b8251> in <module>
     15 )
     16 
---> 17 tuner.run()
     18 
     19 print("Best score:", tuner.best_score)

/usr/local/lib/python3.7/site-packages/optuna/integration/_lightgbm_tuner/optimize.py in run(self)
    461         self.sample_train_set()
    462 
--> 463         self.tune_feature_fraction()
    464         self.tune_num_leaves()
    465         self.tune_bagging()

/usr/local/lib/python3.7/site-packages/optuna/integration/_lightgbm_tuner/optimize.py in tune_feature_fraction(self, n_trials)
    486 
    487         sampler = optuna.samplers.GridSampler({param_name: param_values})
--> 488         self._tune_params([param_name], len(param_values), sampler, "feature_fraction")
    489 
    490     def tune_num_leaves(self, n_trials: int = 20) -> None:

/usr/local/lib/python3.7/site-packages/optuna/integration/_lightgbm_tuner/optimize.py in _tune_params(self, target_param_names, n_trials, sampler, step_name)
    567                 timeout=_timeout,
    568                 catch=(),
--> 569                 callbacks=self._optuna_callbacks,
    570             )
    571 

/usr/local/lib/python3.7/site-packages/optuna/study.py in optimize(self, func, n_trials, timeout, n_jobs, catch, callbacks, gc_after_trial, show_progress_bar)
    290             if n_jobs == 1:
    291                 self._optimize_sequential(
--> 292                     func, n_trials, timeout, catch, callbacks, gc_after_trial, None
    293                 )
    294             else:

/usr/local/lib/python3.7/site-packages/optuna/study.py in _optimize_sequential(self, func, n_trials, timeout, catch, callbacks, gc_after_trial, time_start)
    652                     break
    653 
--> 654             self._run_trial_and_callbacks(func, catch, callbacks, gc_after_trial)
    655 
    656             self._progress_bar.update((datetime.datetime.now() - time_start).total_seconds())

/usr/local/lib/python3.7/site-packages/optuna/study.py in _run_trial_and_callbacks(self, func, catch, callbacks, gc_after_trial)
    683         # type: (...) -> None
    684 
--> 685         trial = self._run_trial(func, catch, gc_after_trial)
    686         if callbacks is not None:
    687             frozen_trial = copy.deepcopy(self._storage.get_trial(trial._trial_id))

/usr/local/lib/python3.7/site-packages/optuna/study.py in _run_trial(self, func, catch, gc_after_trial)
    707 
    708         try:
--> 709             result = func(trial)
    710         except exceptions.TrialPruned as e:
    711             message = "Trial {} pruned. {}".format(trial_number, str(e))

/usr/local/lib/python3.7/site-packages/optuna/integration/_lightgbm_tuner/optimize.py in __call__(self, trial)
    302         cv_results = lgb.cv(self.lgbm_params, self.train_set, **self.lgbm_kwargs)
    303 
--> 304         val_scores = self._get_cv_scores(cv_results)
    305         val_score = val_scores[-1]
    306         elapsed_secs = time.time() - start_time

/usr/local/lib/python3.7/site-packages/optuna/integration/_lightgbm_tuner/optimize.py in _get_cv_scores(self, cv_results)
    292 
    293         metric = self._get_metric_for_objective()
--> 294         val_scores = cv_results["{}-mean".format(metric)]
    295         return val_scores
    296 

KeyError: 'mse-mean'
```

## Steps to reproduce

1. Run [this script](https://github.com/optuna/optuna/blob/master/examples/lightgbm_tuner_cv.py) with objective = regression and metric = mse.

### PR DESCRIPTION
**Pull request by thigm85**
WIP

## Motivation
Fix #1602 

## Description of the changes
Add l2 metric aliases

### CODE REVIEW

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py [hunk 1]
```diff
@@ -352,10 +353,15 @@ def test__parse_args_wrapper_args(self) -> None:
...
+        lgbm_params = {}
+        lgbm_params["metric"] = metric
+        _handling_alias_metrics(lgbm_params)
+        with turnoff_train(metric=lgbm_params["metric"]):
             study = optuna.create_study(direction=study_direction)
             runner = self._get_tuner_object(
-                params=dict(lambda_l1=0.0, metric=metric), kwargs_options={}, study=study,
+                params=dict(lambda_l1=0.0, metric=lgbm_params["metric"]),
+                kwargs_options={},
+                study=study,
```
Comment (toshihikoyanase) [inline tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:364]: Cool! IMO, we have an alternative approach. How about changing the metric name in L353 from `mse` to `l2`? Then, we can keep the existing test code here.

File: optuna/integration/_lightgbm_tuner/alias.py [hunk 2]
```diff
@@ -53,6 +53,10 @@ def _handling_alias_parameters(lgbm_params: Dict[str, Any]) -> None:
         ],
     },
     {"metric_name": "map", "alias_names": ["mean_average_precision"]},
+    {
+        "metric_name": "l2",
+        "alias_names": ["regression", "regression_l2", "l2", "mean_squared_error", "mse"],
+    },
```
Comment (toshihikoyanase) [inline optuna/integration/_lightgbm_tuner/alias.py:59]: Could you add a test case for this case in `tests/integration_tests/lightgbm_tuner_tests/test_alias.py`?
It'll be something like this:

```python
    for alias in ["regression", "regression_l2", "l2", "mean_squared_error", "mse"]:
        lgbm_params = {"metric": alias}
        _handling_alias_metrics(lgbm_params)
        assert lgbm_params["metric"] == "l2"
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. Could you add a test case for the l2 metric alias?

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py (hunk 1 above)
Comment (thigm85) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:364]: I agree. If that is ok I think we should do that. I was not sure if there was a special meaning around the `mse` in that case.

File: optuna/integration/_lightgbm_tuner/alias.py (hunk 2 above)
Comment (thigm85) [reply optuna/integration/_lightgbm_tuner/alias.py:59]: Will do.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your swift action. LGTM!

c.f., The items of the alias list can be found in this [configuration](https://github.com/microsoft/LightGBM/blob/083b02af73b8fcf9a92af78612c6cca6243df752/include/LightGBM/config.h#L1111-L1112).

**Review by HideakiImamura — APPROVED** [review]
Thanks for the PR! LGTM!

**Comment by toshihikoyanase** [comment]
Let me slightly modify the title to show in the release note.

**Pull request merged**